crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -851,7 +870,10 @@ where
                 // TODO: Definitions created by loop variables
                 // (and definitions created inside the body)
                 // are fully visible to other statements/expressions inside the body --Alex/Carl
+                let in_nested_loop = self.inside_loop;
+                self.inside_loop = true;
```

Comment:
We need to set the flag to `true` for while statements

Reply:
I don't understand? We already do?